tests/suite/longevity_test.go

```diff
@@ -0,0 +1,137 @@
+		}
+	})
+
+	It("sets up the longevity test", Label("longevity-setup"), func() {
+		if !strings.Contains(labelFilter, "longevity-setup") {
+			Skip("'longevity-setup' label not specified; skipping...")
+		}
+
+		Expect(resourceManager.Apply([]client.Object{ns})).To(Succeed())
+		Expect(resourceManager.ApplyFromFiles(files, ns.Name)).To(Succeed())
+		Expect(resourceManager.ApplyFromFiles(promFile, ngfNamespace)).To(Succeed())
+		Expect(resourceManager.WaitForAppsToBeReady(ns.Name)).To(Succeed())
+	})
+
+	It("collects results", Label("longevity-teardown"), func() {
+		if !strings.Contains(labelFilter, "longevity-teardown") {
+			Skip("'longevity-teardown' label not specified; skipping...")
+		}
+
+		resultsDir, err := framework.CreateResultsDir("longevity", version)
+		Expect(err).ToNot(HaveOccurred())
+
+		filename := filepath.Join(resultsDir, fmt.Sprintf("%s.md", version))
+		resultsFile, err := framework.CreateResultsFile(filename)
+		Expect(err).ToNot(HaveOccurred())
+		defer resultsFile.Close()
+
+		Expect(framework.WriteSystemInfoToFile(resultsFile, clusterInfo, *plusEnabled)).To(Succeed())
+
+		// gather wrk output
+		homeDir, err := os.UserHomeDir()
+		Expect(err).ToNot(HaveOccurred())
+
+		Expect(framework.WriteContent(resultsFile, "\n## Traffic\n"))
+		writeTrafficResults(resultsFile, homeDir, "coffee.txt", "HTTP")
+		writeTrafficResults(resultsFile, homeDir, "tea.txt", "HTTPS")
+
+		// gather any error logs
+		names, err := framework.GetReadyNGFPodNames(k8sClient, ngfNamespace, releaseName, timeoutConfig.GetTimeout)
+		Expect(err).ToNot(HaveOccurred())
+
+		Expect(framework.WriteContent(resultsFile, "\n## Error Logs\n"))
+		writeErrorLogs(resultsFile, names[0], "nginx-gateway")
+		writeErrorLogs(resultsFile, names[0], "nginx")
+
+		Expect(resourceManager.DeleteFromFiles(files, ns.Name)).To(Succeed())
+		Expect(resourceManager.Delete([]client.Object{ns})).To(Succeed())
+	})
+})
+
+func writeTrafficResults(resultsFile *os.File, homeDir, filename, testname string) {
+	file := fmt.Sprintf("%s/%s", homeDir, filename)
+	content, err := os.ReadFile(file)
+	Expect(err).ToNot(HaveOccurred())
+
+	formattedContent := fmt.Sprintf("%s:\n\n```text\n%s```\n", testname, string(content))
+	Expect(framework.WriteContent(resultsFile, formattedContent)).To(Succeed())
+}
+
+func writeErrorLogs(resultsFile *os.File, pod, container string) {
+	logReq := clientGoClient.CoreV1().Pods(ngfNamespace).GetLogs(pod, &core.PodLogOptions{Container: container})
+
+	ctx, cancel := context.WithTimeout(context.Background(), timeoutConfig.GetTimeout)
+	defer cancel()
+
+	logs, err := logReq.Stream(ctx)
+	Expect(err).ToNot(HaveOccurred())
+	defer logs.Close()
+
+	buf := new(bytes.Buffer)
+	_, err = io.Copy(buf, logs)
+	Expect(err).ToNot(HaveOccurred())
+
+	Expect(framework.WriteContent(resultsFile, fmt.Sprintf("\n### %s\n", container)))
+
+	scanner := bufio.NewScanner(strings.NewReader(buf.String()))
+	for scanner.Scan() {
```

Comment:
Yeah, it could. Won't know for sure until we actually run the real thing. We could verify at that time and think of a new approach if it does take awhile.